### PR TITLE
px sdk update to mesosphere/dcos-commons-0.56.1 

### PR DESCRIPTION
**Pull request by tusharrautOO7**

### CODE REVIEW

File: sdk/scheduler/src/test/java/com/mesosphere/sdk/testutils/TestPodFactory.java [hunk 1]
```diff
@@ -131,7 +131,8 @@ public static ResourceSet getResourceSet(String id, double cpu, double mem, doub
                 .id(id)
                 .cpus(cpu)
                 .memory(mem)
-                .addRootVolume(disk, TestConstants.CONTAINER_PATH)
+                .addVolume(VolumeSpec.Type.ROOT.toString(), "", "", null, disk, TestConstants.CONTAINER_PATH, Collections.emptyList())
+                //.addRootVolume(disk, TestConstants.CONTAINER_PATH)
```
Comment (disrani-px) [inline sdk/scheduler/src/test/java/com/mesosphere/sdk/testutils/TestPodFactory.java]: Remove commented code

File: frameworks/elastic/src/test/java/com/mesosphere/sdk/elastic/scheduler/ServiceTest.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ public void testRegionAwareness() throws Exception {
     ServiceTestResult result = getDefaultRunner()
         .setOptions("service.region", "Europe")
         .run();
-    Assert.assertEquals(result.getSchedulerEnvironment().get("SERVICE_REGION"), "Europe");
+    Assert.assertEquals(result.getSchedulerEnvironment().get("SERVICE_REGION"), null);
```
Comment (disrani-px) [inline frameworks/elastic/src/test/java/com/mesosphere/sdk/elastic/scheduler/ServiceTest.java]: Undo this change

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/strategy/SerialStrategy.java [hunk 3]
```diff
@@ -35,27 +35,19 @@ public String getName() {
...
-          .filter(el -> !el.isComplete())
-          .collect(Collectors.toList());
-      Collections.reverse(planElements);
+    dependencyStrategyHelper = new DependencyStrategyHelper<>(elements);
+    List<C> planElements = elements.stream()
+        .filter(el -> !el.isComplete())
+        .collect(Collectors.toList());
+    Collections.reverse(planElements);
 
-      // Note: We mark ALL dependencies (including inferred dependencies) because DependencyStrategyHelper doesn't
```
Comment (disrani-px) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/strategy/SerialStrategy.java:45]: Don't remove comments

File: tools/release_builder.py [hunk 4]
```diff
@@ -62,9 +62,8 @@ def apply_beta_prefix(package_name: str, is_beta: bool) -> str:
 
     @staticmethod
     def apply_beta_version(package_version: str, is_beta: bool) -> str:
-        """Add the '-beta' suffix to the package version if required"""
-
-        stripped_version = right_trim(package_version, "-beta")
+        '''Add the '-beta' suffix to the package version if required'''
```
Comment (disrani-px) [inline tools/release_builder.py]: Don't need this change anymore

File: frameworks/portworx/universe/marathon.json.mustache [hunk 5]
```diff
@@ -33,13 +33,15 @@
     "FRAMEWORK_NAME": "{{service.name}}",
     "FRAMEWORK_PRINCIPAL": "{{service.principal}}",
     "PRE_RESERVED_ROLE": "{{service.pre_reserved_role}}",
+    "TASK_FAILURE_TIMEOUT_MINUTES": "{{{service.task_failure_timeout_minutes}}}",
```
Comment (disrani-px) [inline frameworks/portworx/universe/marathon.json.mustache:35]: This change is missing the update in `Main.java` for where `TASK_FAILURE_TIMEOUT_MINUTES` is being used?
Commit 17bb8c03c9874140e29b2e7f6797c3275210d680 from px_1.3-0.40.5

File: frameworks/elastic/src/main/dist/svc.yml [hunk 6]
```diff
@@ -48,7 +48,16 @@ pods:
           docker_volume_driver: pxd
           docker_volume_name: {{{MASTER_NODE_DOCKER_VOLUME_NAME}}}
           docker_driver_options: {{{MASTER_NODE_DOCKER_DRIVER_OPTIONS}}}
-        cmd: "export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jdk*/jre/) && ./elastic-scheduler/java-dns-ttl.sh && ./elastic-scheduler/install-plugins.sh && && ./elastic-scheduler/setup-java-policy.sh && ./bootstrap && sudo chown {{FRAMEWORK_USER}} $MESOS_SANDBOX/container-path -R && exec ./elasticsearch-{{ELASTIC_VERSION}}/bin/elasticsearch"
+        cmd: |
+          set -ex
```
Comment (disrani-px) [inline frameworks/elastic/src/main/dist/svc.yml:52]: All the commands are missing `sudo chown {{FRAMEWORK_USER}} $MESOS_SANDBOX/container-path -R` which will cause the framework to fail

File: testing/sdk_metrics.py [hunk 7]
```diff
@@ -171,14 +171,10 @@ def get_metrics(package_name: str, service_name: str, pod_name: str, task_name:
...
     app_response = sdk_cmd.cluster_request(
-        "GET",
-        "/system/v1/agent/{}/metrics/v0/containers/{}/app".format(
-            task_to_check.agent_id, task_container_id
-        ),
-        retry=False,
-    )
-    app_response.raise_for_status()
-    app_json = app_response.json()
+        "GET", "/system/v1/agent/{}/metrics/v0/containers/{}/app".format(agent_id, task_container_id), retry=False)
```
Comment (disrani-px) [inline testing/sdk_metrics.py]: Undo commit 930f82fad9f850d40942edc1b4c4d09253f1aeb7, it's no longer required

File: testing/sdk_metrics.py [hunk 8]
```diff
@@ -210,6 +204,19 @@ def check_metrics_presence(emitted_metrics: List[str], expected_metrics: List[st
 
     return True
 
+    missing_metrics = []
```
Comment (disrani-px) [inline testing/sdk_metrics.py]: Undo commit ae166cbca4dbcf39ea643ae4613d2a1a5a597daa, it is no longer required

**Review by disrani-px — CHANGES_REQUESTED** [review]
These commits are empty: 
https://github.com/portworx/dcos-commons/pull/128/commits/17d72dfd7ce695b885427db3b23610f779eb50ca
https://github.com/portworx/dcos-commons/pull/128/commits/2d2929ed9f1d6e72a745b3e60bb438cb1c880225
https://github.com/portworx/dcos-commons/pull/128/commits/139e6a2ead8cc09c16591b04b65c5075233e4665
https://github.com/portworx/dcos-commons/pull/128/commits/4ab8283ba5310d90b9ab9023e5f8186da5bd3c0e
https://github.com/portworx/dcos-commons/pull/128/commits/ef09a9c5be1bfb6fbed3956e4908e4caf406ed28
https://github.com/portworx/dcos-commons/pull/128/commits/2deb31c6d184f7427d277c8412060c8607afae12
https://github.com/portworx/dcos-commons/pull/128/commits/89bbde53a5d97ddbae2ce04b6401e4c8db2f71ed
https://github.com/portworx/dcos-commons/pull/128/commits/08ea635b32e301fd81c856db16fc1dba0d28078c
https://github.com/portworx/dcos-commons/pull/128/commits/27dbed3d91f85914cf2dd216af2ab94ed530b360
https://github.com/portworx/dcos-commons/pull/128/commits/23531df0855334c6db78971bdcd4b87b4586b282
https://github.com/portworx/dcos-commons/pull/128/commits/aa5af84c82cb3d6d36caa2a31679bc8c0307da69
https://github.com/portworx/dcos-commons/pull/128/commits/74a7583b575c03272721cb9e7e3149c8809803e8
https://github.com/portworx/dcos-commons/pull/128/commits/efe20fc391569421b3e77c82e5d98369d5c5421b
https://github.com/portworx/dcos-commons/pull/128/commits/f0589e59b030425103892dfc28e6c2c14213e1d6

This change has a merge conflict marker: https://github.com/portworx/dcos-commons/pull/128/commits/c6f6ab47a19d80d8dc05fe405bc019a4da37678a

Also remove the empty kafka commits

File: tools/release_builder.py (hunk 4 above)
Comment (tusharrautOO7) [reply tools/release_builder.py]: Changed.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/strategy/SerialStrategy.java (hunk 3 above)
Comment (tusharrautOO7) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/strategy/SerialStrategy.java:45]: Added changes.

File: frameworks/elastic/src/test/java/com/mesosphere/sdk/elastic/scheduler/ServiceTest.java (hunk 2 above)
Comment (tusharrautOO7) [reply frameworks/elastic/src/test/java/com/mesosphere/sdk/elastic/scheduler/ServiceTest.java]: Changed.

File: testing/sdk_metrics.py (hunk 8 above)
Comment (tusharrautOO7) [reply testing/sdk_metrics.py]: Done.

File: testing/sdk_metrics.py (hunk 7 above)
Comment (tusharrautOO7) [reply testing/sdk_metrics.py]: Done.

**Pull request closed without merging**